crates/subspace-core-primitives/src/lib.rs

```diff
@@ -82,6 +82,30 @@ pub const BLAKE3_HASH_SIZE: usize = 32;
 /// BLAKE3 hash output
 pub type Blake3Hash = [u8; BLAKE3_HASH_SIZE];
 
+/// BLAKE3 hash output wrapper, which serializes it as a hex string
```

Comment:
I would add a TODO that this is a temporary hack and `Blake3Hash` itself should be converted into new type instead

Reply:
```suggestion
/// BLAKE3 hash output wrapper, which serializes it as a hex string
// TODO: rename this type to Blake3Hash into a newtype, after checking for any breaking changes
```

Reply:
There was a typo so I pushed commit [dd777f2](https://github.com/autonomys/subspace/pull/2996/commits/dd777f24cdd9476cc85744956064080a51c3b041) instead.